back/engines/commercial/nlp/lib/nlp/text_network.rb

```diff
@@ -1,22 +1,107 @@
 # frozen_string_literal: true
+
 require 'active_support/core_ext/hash/indifferent_access'
 
 module NLP
   class TextNetwork
-    attr_reader :nodes, :links, :communities, :directed
+    class << self
+      def from_json(json_network)
+        json_network = JSON.parse(json_network) if json_network.is_a?(String)
+
+        text_network = json_network.fetch('text_network')
+        nodes = text_network['nodes'].map { |json_node| Node.new(json_node['id'], json_node['pagerank']) }
+
+        new(nodes, directed: text_network['directed']).tap do |network|
+          # Adding links
+          text_network.fetch('links', []).each do |json_link|
+            network.add_link(json_link['source'], json_link['target'], json_link['weight'])
+          end
+
+          # Adding communities
+          json_network.fetch('communities', []).each do |json_community|
+            network.add_community(
+              json_community['partitions_id'],
+              json_community['influent_nodes'],
+              json_community['relative_size']
+            )
+          end
+        end
+      end
+
+      def merge(*networks)
+        is_directed = merge_directed?(networks)
+        nodes = networks.flat_map(&:nodes)
+        links = networks.flat_map(&:links)
+        communities = networks.flat_map(&:communities)
+
+        raise 'Node identifier collision' if duplicates?(nodes.map(&:id))
```

Comment:
Isn't the name-spacing avoiding this already ?

Reply:
Yep, but `merge` and `namespace` are somehow independent from each other. Here we are just making sure that the resulting network is valid and does not contain duplicates. So if u will, it checks if the networks were indeed appropriately namespaced before -- even if in practice, merge does not really care as long as there are no collisions.